lib/full-site-editing/template-loader.php

```diff
@@ -266,3 +266,28 @@ function gutenberg_resolve_template_for_new_post( $wp_query ) {
 		$wp_query->set( 'post_status', 'auto-draft' );
 	}
 }
+
+/**
+ * Redirect the edit links for templates to the site editor.
+ *
+ * @param string $link    The original link.
+ * @param int    $post_id The custom post id.
+ */
+function gutenberg_get_edit_template_link( $link, $post_id ) {
+	$post = get_post( $post_id );
+
+	if ( ! in_array( $post->post_type, array( 'wp_template', 'wp_template_part' ), true ) ) {
+		return $link;
+	}
+
+	$template = _build_block_template_result_from_post( $post );
+
+	if ( is_wp_error( $template ) ) {
+		return $link;
+	}
+
+	$edit_link = 'themes.php?page=gutenberg-edit-site&postId=%1$s&postType=%2$s';
```

Comment:
Shouldn't we use [`build_query`](https://developer.wordpress.org/reference/functions/build_query/) to build `$edit_link`?
cc @Mamaduka

Reply:
I think it's a simple enough query that there's no need to use `build_query` or `add_query_arg.`

P.S. In the future, this path can be a part of a post type of object via the `_edit_link` argument.

Reply:
There is a library function for building queries, and it feels like we are inventing the wheel here.
But I'm not going to insist we need to change it because we encode the request parameters (as it should be).